src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -69,19 +77,54 @@ def evaluate_workflow_triggers(workflows: set[Workflow], job: WorkflowJob) -> se
     workflows_to_enqueue: set[Workflow] = set()
 
     for workflow in workflows:
-        if workflow.evaluate_trigger_conditions(job):
-            triggered_workflows.add(workflow)
+        evaluation, remaining_conditions = workflow.evaluate_trigger_conditions(job)
+        if remaining_conditions:
+            workflows_to_enqueue.add(workflow)
         else:
-            if get_slow_conditions(workflow):
-                # enqueue to be evaluated later
-                workflows_to_enqueue.add(workflow)
+            if evaluation:
+                # Only add workflows that have no remaining conditions to check
+                triggered_workflows.add(workflow)
 
     if workflows_to_enqueue:
         enqueue_workflows(workflows_to_enqueue, job)
 
     return triggered_workflows
 
 
+def evaluate_workflows_action_filters(
+    workflows: set[Workflow],
+    job: WorkflowJob,
+) -> BaseQuerySet[Action]:
+    filtered_action_groups: set[DataConditionGroup] = set()
+    enqueued_conditions: list[DataCondition] = []
+
+    # gets the list of the workflow ids, and then get the workflow_data_condition_groups for those workflows
+    workflow_ids = {workflow.id for workflow in workflows}
+
+    action_conditions = DataConditionGroup.objects.filter(
+        workflowdataconditiongroup__workflow_id__in=workflow_ids
+    ).distinct()
+
+    for action_condition in action_conditions:
+        evaluation, result, remaining_conditions = evaluate_condition_group(action_condition, job)
+
+        if remaining_conditions:
+            # If there are remaining conditions for the action filter to evaluate,
+            # then return the list of conditions to enqueue
+            enqueued_conditions.extend(remaining_conditions)
```

Comment:
i had planned to enqueue the `DataConditionGroup`, i think we can still do that here because the `remaining_conditions` can only be slow conditions, or am i missing something about why it's better to enqueue the conditions themselves? or is this more generic in case `remaining_conditions` might not only be slow conditions in the future?

Reply:
on second thought, probably need to rethink the enqueuing logic since i am planning to have the key be `{workflow_id}:{group_id}:{dcg_id,...,dcg_id}`

Reply:
yeah, i ended up changing the enqueue method a bit:
https://github.com/getsentry/sentry/blob/67d137ec4328b5808ea614cb9b2073e33cc74117/src/sentry/workflow_engine/processors/workflow.py#L32-L52

Since each condition knows it's condition group, i just made the enqueue method signature a little more ergonomic

Reply:
yeah this works. will have to refactor my PR using this 😅

Reply:
out of curiosity, why's that? aren't you just reading from the buffer and that's all the format we chatted about this mornin'

Reply:
i was assuming i would get the list of `DataConditionGroup` ids and work from there, but here we are putting `DataCondition` ids + the kind of `DataConditionGroup` (trigger or filter) instead. i'll have to parse that a little differently now

Reply:
oh jk i need to learn to read

Reply:
i think i was misled that we are passing the conditions be enqueued when we're still actually enqueuing the 
`DataConditionGroup`, do we need to collect all the conditions in that case?

Reply:
eh, probably don't need them all, we could probably move that list coercion up and out of this method and just take a list of ids in the method signature, but can figure that out on the other PR :)